### PR TITLE
Push-down compare function for Dictionary and RunEnd

### PR DESCRIPTION
**Pull request by gatesn**

### CODE REVIEW

File: encodings/dict/src/compute.rs [hunk 1]
```diff
@@ -18,9 +29,32 @@ impl ArrayCompute for DictArray {
...
+impl MaybeCompareFn for DictArray {
+    fn maybe_compare(
+        &self,
+        other: &ArrayData,
+        operator: Operator,
+    ) -> Option<VortexResult<ArrayData>> {
+        // If the RHS is constant, then we just need to compare against our encoded values.
+        if other
+            .statistics()
+            .get_as::<bool>(Stat::IsConstant)
```
Comment (robert3005) [inline encodings/dict/src/compute.rs:43]: do you want a get or do you want a compute here?

Comment (gatesn) [reply encodings/dict/src/compute.rs:43]: Almost always a get I think.

**Review by robert3005 — APPROVED** [review]

File: encodings/dict/src/compute.rs (hunk 1 above)
Comment (lwwmanning) [reply encodings/dict/src/compute.rs:43]: Yeah, on ConstantArrays we set IsConstant in the constructor

**Review by lwwmanning — COMMENTED** [review]
Write a test for each of these new functions please

**Pull request merged**